walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -20,15 +20,30 @@
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    ArrayList<TimeRange> conflicts = getConflictingEvents(events, request);
-    return getPossibleTimeRanges(conflicts, request.getDuration());
+    Collection<String> mandatoryAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    long duration = request.getDuration();
+
+    ArrayList<String> allAttendees = new ArrayList<String>();
+    allAttendees.addAll(mandatoryAttendees);
+    allAttendees.addAll(optionalAttendees);
+
+    ArrayList<TimeRange> allConflicts = getConflictingEvents(events, allAttendees);
+    ArrayList<TimeRange> possibleTimerangesAllAttendees =
+        getPossibleTimeRanges(allConflicts, duration);
+    if (possibleTimerangesAllAttendees.isEmpty() && !mandatoryAttendees.isEmpty()) {
+      ArrayList<TimeRange> mandatoryConflicts =
+          getConflictingEvents(events, request.getAttendees());
```

Comment:
Nit: don't call `request.getAttendeesI()` again here. Reuse `mandatoryAttendees`.

In general if you call a method and store the result in a variable, reuse the variable throughout the function. That makes it clearer where you're reusing a result you've used before.

Reply:
Oh you're right! I did that while debugging and forgot to change it. Great catch!
```suggestion
          getConflictingEvents(events, mandatoryAttendees);
```